packages/stylelint-plugin-mozaic/lib/checkCase.js

```diff
@@ -0,0 +1,45 @@
+var stylelint = require('stylelint')
+
+const cases = {
+  'kebab-case': /^([a-z][a-z0-9]*)(-[a-z0-9]+)*$/,
+  lowerCamelCase: /^[a-z][a-zA-Z0-9]+$/,
+  snake_case: /^[a-z][a-zA-Z0-9]+$/,
+  PascalCase: /^[A-Z][a-zA-Z0-9]+$/,
+  UpperCamelCase: /^[A-Z][a-zA-Z0-9]+$/,
+}
+const messages = ''
+
+const match = (string, configCase) => cases[configCase].test(string)
+
+const checkCase = (splitedSelector, rule, result, ruleName, options) => {
+  let isValid = true
+
+  splitedSelector.forEach(selectorPart => {
+    if (selectorPart.type === 'class') {
+      selectorPart.bemStructure.forEach(classPart => {
+        // ignore # for variables in scss files
+        const isValidCase = classPart.string.includes('#')
+          ? true
+          : match(classPart.string, options.wordDelimiterStyle)
+
+        if (!isValidCase) {
+          isValid = false
+        }
+      })
+    }
+  })
+
+  if (!isValid) {
+    message = `Invalid class syntax: "${rule.selector}" should be "${options.wordDelimiterStyle}" (${ruleName})`
```

Comment:
we need to give a link to users to the doc related to the error. 
we should have a page listing all errors in the docs.